dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/sms/SmsGatewayController.java

```diff
@@ -114,62 +112,62 @@ public void getGatewayConfiguration( @PathVariable String uid, HttpServletRespon
 
     @PreAuthorize( "hasRole('ALL') or hasRole('F_MOBILE_SENDSMS')" )
     @PutMapping( "/default/{uid}" )
-    public void setDefault( @PathVariable String uid, HttpServletRequest request, HttpServletResponse response )
-        throws WebMessageException
+    @ResponseBody
+    public WebMessage setDefault( @PathVariable String uid )
     {
         SmsGatewayConfig gateway = gatewayAdminService.getByUid( uid );
 
         if ( gateway == null )
         {
-            throw new WebMessageException( WebMessageUtils.notFound( "No gateway found" ) );
+            return WebMessageUtils.notFound( "No gateway found" );
         }
 
         gatewayAdminService.setDefaultGateway( gateway );
 
-        webMessageService.send( WebMessageUtils.ok( gateway.getName() + " is set to default" ), response, request );
+        return WebMessageUtils.ok( gateway.getName() + " is set to default" );
     }
 
     @PreAuthorize( "hasRole('ALL') or hasRole('F_MOBILE_SENDSMS')" )
     @PutMapping( "/{uid}" )
-    public void updateGateway( @PathVariable String uid, HttpServletRequest request, HttpServletResponse response )
-        throws WebMessageException,
-        IOException
+    public WebMessage updateGateway( @PathVariable String uid, HttpServletRequest request )
+        throws IOException
     {
         SmsGatewayConfig config = gatewayAdminService.getByUid( uid );
 
         if ( config == null )
         {
-            throw new WebMessageException( WebMessageUtils.notFound( "No gateway found" ) );
+            return WebMessageUtils.notFound( "No gateway found" );
         }
 
         SmsGatewayConfig updatedConfig = renderService.fromJson( request.getInputStream(), SmsGatewayConfig.class );
 
-        if ( gatewayAdminService.hasDefaultGateway() && updatedConfig.isDefault() )
+        if ( gatewayAdminService.hasDefaultGateway() && updatedConfig.isDefault() && !config.isDefault() )
```

Comment:
Note that adding `&& !config.isDefault()` is a fix so that the default config can be updated.